web/spec/supabase.yml

```diff
@@ -709,6 +705,35 @@ pages:
           const { data, error } = await supabase.auth.api
             .resetPasswordForEmail('user@email.com')
           ```
+      - name: React example
+        isSpotlight: false
+        js: |
+          ```js
+          /**
+           * Step 1: Send the user an email to get a password reset token.
+           *   This email will contain a link which will send the user back to your application.
+           */
+          const { data, error } = await supabase.auth
+            .resetPasswordForEmail('user@email.com')
```

Comment:
@kangmingtay do you know if a dev can provide a `redirectTo` option here?

```js
const { data, error } = await supabase.auth
            .resetPasswordForEmail('user@email.com', { redirectTo: 'XXX' })
```

Reply:
yup they can (see [here](https://github.com/supabase/gotrue-js/blob/master/src/GoTrueApi.ts#L439)) also it should be `supabase.auth.api.resetPasswordForEmail` since this should only be called on the server-side

Reply:
Hmm yeah actually it doesn't require any sensitive tokens, it doesn't have to be called with the user's access token as well (since they wouldn't be able to login in the first place if they are requesting for a recovery link) - they would just need to call this with the anon key.

If captcha is enabled in the auth settings page, then the developer will have to add the captcha challenge on the front-end as well. This captcha token refers to the verification token the user gets after completing the captcha challenge on the site.

Reply:
> If captcha is enabled in the auth settings page, then the developer will have to add the captcha challenge on the front-end as well

do you have any more docs on this other than what is written in the [config reference](https://supabase.com/docs/reference/tools/reference-auth)?

<img width="940" alt="image" src="https://user-images.githubusercontent.com/10214025/176014590-3cbb2e23-a51b-4e4d-8ed8-406885e88908.png">

How do I get a captcha code? Are there certain captcha providers I can use?

Reply:
1. we only support the hCaptcha provider for now
2. The captcha token is obtained after the user completes a series of "challenges" (e.g. identifying which pictures are pictures of a car)
3. hCaptcha has [some sample javascript](https://docs.hcaptcha.com/configuration) to render the captcha widget - i'm not too sure how we can improve the side of the DX for our users yet but it'll be awesome if we have an auth component that renders the captcha widget for them automatically